Updating Documentation Steps and Formatting with Additional Details (#211)
agera-edc/MinimumViableDataspace: system-tests/README.md

# System tests

The system tests copy a file from a provider to a consumer blob storage account.

## Publish/Build Tasks

> ! Important Note !
> 
> MVD dependencies are Eclipse DataSpaceConnector(EDC) and Registration Service. Both of these dependencies 

> > are __not__ published to any central artifactory yet, so in local development we have to use locally 

Reviewer: artifactory is a product name
```suggestion
> are __not__ published to any central artifact repository yet, so in local development we have to use locally 
```

# System tests

The system tests copy a file from a provider to a consumer blob storage account.

> ## Publish/Build Tasks

Reviewer 2: THe MVD on `main` now also uses the IdentityHub, which should be included in the instructions as well.

Author: at this stage the docker-compose for the local deployment is relying on `nginx`, when the local deploy is updated, the doc should be updated to reflect  this

>
> For Example, the `Registration Service` and `EDC` repository hash can be found in the _Checkout_ steps  (in the `ref` property) of [action.yml](./.github/actions/../../../.github/actions/gradle-setup/action.yml):

```yml
    - name: Checkout EDC
      uses: actions/checkout@v2
      with:
        repository: eclipse-dataspaceconnector/DataSpaceConnector
        path: DataSpaceConnector
        ref: 3ff940b720f44826df28e893fb31344eb6faacef

    - name: Checkout Registration Service
      uses: actions/checkout@v2
      with:
        repository: eclipse-dataspaceconnector/RegistrationService
        path: RegistrationService
        ref: 374c14bcca23ddb1dcd7476a27264510e54de7fa
```

> After you have cloned the `EDC` and `Registration Service` repos locally you should run the command to
> `reset` to the specific hash.  
>
> For Example:

```bash
# EDC (in the EDC root folder)
git reset --hard 3ff940b720f44826df28e893fb31344eb6faacef

# Registration Service (in the Registration Service root folder)
git reset --hard 374c14bcca23ddb1dcd7476a27264510e54de7fa
```

> Now you can follow the rest of the process below.  

> > Once the publications are available in _Maven Central_ this process will not be necessary

Reviewer 2: Artifacts should now be available according to Paul's post in Discord.

Author: yes but at this time, not building with the current version of the MVD, we should follow up with this at a later stage and when all is confirmed, make a task to updated the documentation

> 
<br />

### EDC

<br />

> Checkout [Eclipse DataSpaceConnector repository](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector). Next, publish `EDC` libraries to local Maven artifactory by executing the following command  from `EDC` root folder.

Reviewer 3: nit: maybe we can remove this checkout step now as it is already covered in previous section, Thoughts?

Same for RegistrationService section below.

Author: I will update the wording, the intention here is that when we are good to go with `main` repo's it can be checked out, next stage would be working with the OSSRH hosted repos

- `MVD` runs three `EDC Connectors` and one `Registration Service`.

_Note: Ensure that you are able to build `MVD` locally as described in the previous [section](#mvd)._

First, we need to build `EDC Connector launcher` and `Registration Service launcher`.

From the `MVD` root folder, execute the following command:

```bash
./gradlew -DuseFsVault="true" :launcher:shadowJar
```

From the `Registration Service` root folder, execute the following command:

```bash
./gradlew :launcher:shadowJar
```

From the `MVD` root folder execute the following commands to set the `Registration Launcher` path environment variable and start `MVD` using the `docker-compose.yml` file.

```bash
export REGISTRATION_SERVICE_LAUNCHER_PATH=/home/user/RegistrationService/launcher
docker-compose -f system-tests/docker-compose.yml up --build
```

_Note for Windows PowerShell, the following commands should be used from the `MVD` project root.  (The path will depend on the location of the your `RegistrationService` project root):_

```powershell
$Env:REGISTRATION_SERVICE_LAUNCHER_PATH="c:/RegistrationService/launcher"
docker-compose -f system-tests/docker-compose.yml up --build
```

> Once completed, these commands will start:

Reviewer 3: nit: maybe we can say docker containers.
```suggestion
Once completed, following services will start within their docker containers:
```

- 3 `EDC Connectors`
  - _consumer-us_
  - _consumer-eu_
  - _provider_
- A `Registration Service`
- A `HTTP Nginx Server` (to serve DIDs) 

> - An `Azurite` blob storage service which will also be seeded with initial required data using a [postman collection](../deployment/data/MVD.postman_collection.json).

Reviewer 3: provider will be seeded with initial data not Azurite.

# Register Consumer-EU
java -jar $Env:REGISTRATION_SERVICE_CLI_JAR_PATH -s="http://localhost:8184/api" participants add --request="{ \`"name\`": \`"consumer-eu\`", \`"supportedProtocols\`": [ \`"ids-multipart\`" ], \`"url\`": \`"http://consumer-eu:8282\`" }"

# Register Consumer-US
java -jar $Env:REGISTRATION_SERVICE_CLI_JAR_PATH -s="http://localhost:8184/api" participants add --request="{ \`"name\`": \`"consumer-us\`", \`"supportedProtocols\`": [ \`"ids-multipart\`" ], \`"url\`": \`"http://consumer-us:8282\`" }"
```

Set the environment variable `TEST_ENVIRONMENT` to `local` to enable local blob transfer test and then run `MVD` system test using the following command:

```bash
export TEST_ENVIRONMENT=local
./gradlew :system-tests:test
```

_Note for Windows PowerShell, the following commands should be used:_

```powershell
$Env:TEST_ENVIRONMENT = "local"
./gradlew :system-tests:test
```

> [Storage Explorer](https://azure.microsoft.com/features/storage-explorer/) can be used to connect to the `Azurite` storage container on `127.0.0.1:10000` port and under `consumereuassets`, account transferred blob can be viewed.

### Local Test Resources

> The following test resources are provided in order to run `MVD` locally. `system-tests/docker-compose.yml` usages it to start `MVD`.

Reviewer 2: ```suggestion
The following test resources are provided in order to run `MVD` locally. `system-tests/docker-compose.yml` uses it to start `MVD`.
```

Not 100% sure if that's the intended wording, please double-check.
